### PR TITLE
Variomedia DNS API

### PR DESCRIPTION
**Pull request by peterkelm**
Initial commit for the Variomedia DNS API implementation.
Note that this hasn't been tested against wildcard certificates We are currently reviewing feasibility with the API.

### CODE REVIEW

**Comment by peterkelm** [comment]
Fixed dns_variomedia_rm to respect the txtvalue supplied as function parameter and only delete the corresponding record. Tested against wildcard record generation - works as expected.

**Comment by peterkelm** [comment]
@Neilpang, Any hints on how to fixt the failing Travis CI check?

File: dnsapi/dns_variomedia.sh [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  _debug _domain "$_domain"
+
+  _debug 'Getting txt records'
+
+  if ! _variomedia_rest GET "dns-records?filter[domain]=$_domain"; then
+    _err 'Error'
+    return 1
+  fi
+
+  _record_id="$(echo $response | cut -d '[' -f2 | cut -d']' -f1  | sed 's/},[ \t]*{/\},§\{/g' | tr § '\n' | grep $_sub_domain | grep $txtvalue | sed 's/^{//;s/}[,]?$//' | tr , '\n' | tr -d '\"' | grep ^id | cut -d : -f2 | tr -d ' ')"
```
Comment (Neilpang) [inline dnsapi/dns_variomedia.sh]: here is 2 spaces,  `| cut -d']' -f1  |`
please remove one as `| cut -d']' -f1 |`

Comment (Neilpang) [reply dnsapi/dns_variomedia.sh]: please use double quote :  ` | grep "$_sub_domain" |`

File: dnsapi/dns_variomedia.sh [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        _sub_domain="$(echo "$fulldomain" | sed "s/\\.$h\$//")"
+        _domain=$h
+        return 0
+      fi
+    fi
+    i=$(_math "$i" + 1)
+  done
+
+  _debug "root domain not found"
+
```
Comment (Neilpang) [inline dnsapi/dns_variomedia.sh]: remove this empty line

File: dnsapi/dns_variomedia.sh (hunk 1 above)
Comment (peterkelm) [reply dnsapi/dns_variomedia.sh]: Done.

File: dnsapi/dns_variomedia.sh (hunk 2 above)
Comment (peterkelm) [reply dnsapi/dns_variomedia.sh]: Done too.

**Pull request merged**